feat: Expose Revenue Analytics insights/queries to Max

rafaeelaudibert opened the pull request:
That's a lot of work, but Max has access to the Revenue analytics queries in its context now!

Followed the README.md file and updated it slightly because there was a tiny thing missing.

Also, split the `format.py` and `test_format.py` file into several files, should be much more organized now

rafaeelaudibert committed:
feat: Expose Revenue Analytics insights/queries to Max

That's a lot of work, but Max has access to the Revenue analytics queries in its context now!

Followed the README.md file and updated it slightly because there was a tiny thing missing.

Also, split the `format.py` and `test_format.py` file into several files, should be much more organized now

rafaeelaudibert commented on ee/hogai/graph/query_executor/format/test/test_funnel.py:
All of these `test_*` files test cases existed already with the exception of `test_revenue_analytics.py`. They used to be inside `test_format.py` but were extracted to separate files.

rafaeelaudibert commented on ee/hogai/graph/query_executor/format/funnel.py:
All of these `format/*.py` files existed already with the exception of `revenue_analytics.py`. They used to be all inside `format.py` but were extracted to separate files.

rafaeelaudibert committed:
fix: Add extra `breakdown` to tests

rafaeelaudibert committed:
fix: Fix types

github-actions[bot] committed:
Update query snapshots

rafaeelaudibert committed:
refactor: Fix mypy

github-actions[bot] committed:
Update UI snapshots for `chromium` (15)

github-actions[bot] committed:
Update UI snapshots for `chromium` (15)

denakorita commented:
I tried it locally, sadly it seems to not work for me, I believe something's up in the frontend. See the console logs too. 

<img width="1902" height="703" alt="Screenshot 2025-10-01 at 16 13 21" src="https://github.com/user-attachments/assets/f0d2f2d9-04c1-4834-a305-0a96dbcadb0f" />

Error trace:

`TypeError: Cannot read properties of undefined (reading 'toString')
    at http://localhost:8234/src/scenes/max/maxContextLogic.ts?t=1759327862466:309:27
    at Array.forEach (<anonymous>)`

rafaeelaudibert commented on ee/hogai/graph/query_executor/nodes.py:
```diff
@@ -96,8 +109,16 @@ def _get_example_prompt(self, viz_message: VisualizationMessage) -> str:
+        if isinstance(viz_message.answer, AssistantRetentionQuery | RetentionQuery):
             return RETENTION_EXAMPLE_PROMPT
-        if isinstance(viz_message.answer, AssistantHogQLQuery):
+        if isinstance(viz_message.answer, AssistantHogQLQuery | HogQLQuery):
```
Yeah, you were missing it, there were no example prompts when investigating an existing insight, only when investigating an insight created by Max itself, this is a bugfix

rafaeelaudibert committed:
refactor: Use better key

rafaeelaudibert commented:
Ugh, because these aren't real insights fixing the frontend problem is tricky 🤦🏻. I'm trying to come up with something, thanks for the review here @denakorita! Will ping you once this is ready for re-review

rafaeelaudibert committed:
feat: Display pydantic error locally

Rather than swallowing the error and replacing with a validation error, just expose the inner error locally

rafaeelaudibert committed:
feat: Support revenue analytics insights in the UI

This is not the prettiest code, but because revenue analytics insights are not really insights we need to find a way to access them via the `revenueAnalyticsLogic` rather than through the `insightsLogic` we already have.

This does the trick in the simplest way possible while still maintaining the impression to the backend that this is an insight.

rafaeelaudibert committed:
feat: Improve prompts to have max use the proper currency prefix

rafaeelaudibert commented:
Ok, fixed, thanks for pointing that out @denakorita! Big oversight from my part

denakorita reviewed (approved):
Looks good to me!
